spec/lib/slug_spec.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require 'slug'
+
+RSpec.describe Slug do
+  it 'raises an exception generating a slug out of nil' do
+    expect do
+      Slug.generate(nil)
+    end.to raise_error(ArgumentError)
+  end
+
+  it 'does not swallow user input when generating a slug' do
+    expect(Slug.generate('-')).to eq('-')
+  end
+
+  it 'normalizes Unicode characters' do
+    string = 'Càfé'
+    expect(
+      Slug.generate(string.unicode_normalize(:nfd))
+    ).to eq('càfé'.unicode_normalize(:nfkc))
+  end
+
+  it 'folds ASCII characters down' do
+    expect(Slug.generate('HELLO')).to eq('hello')
+  end
+
+  it 'folds Unicode characters down' do
+    expect(Slug.generate('ĲLAND')).to eq('ĳland')
+    expect(Slug.generate('Århüs')).to eq('århüs')
```

Comment:
ActiveSupport::Inflector#parameterize "transliterates" all accented characters to ASCII — do you think there's a valid reason why they do that?

Reply:
As an aside, that method has absolutely no place on the inflector. I tried to protest it, but I lost.

No, that was just a misunderstanding of the specs combined with bad implementations in ancient browser versions. Since RFC3986 you can use any UTF-8 character in a URL but you have to percent-encode it. This is exactly what Rails does.

A permalink database value `Ĳ` would have to be encoded as: `/someone/playlists/%C4%B2`, but a modern browser will present it as `/someone/playlists/Ĳ`.

---

Same goes for badgers.

<img width="446" alt="Screenshot 2019-11-11 at 17 52 20" src="https://user-images.githubusercontent.com/1195/68605080-1a477c80-04ac-11ea-825e-dac086cf988b.png">

<img width="446" alt="Screenshot 2019-11-11 at 17 53 34" src="https://user-images.githubusercontent.com/1195/68605153-377c4b00-04ac-11ea-8e02-b860a1dc21b4.png">

Reply:
> a modern browser

Do we know how modern is modern? I'm not too worried about this, but I'm usually so cavalier that it might be nice to know what we are cutting off.

Reply:
My only real concern is maintenance burden, i.e. how likely is it that I'll have to come bug Manfred because we have a custom built UTF-8 friendly permalink scheme that turns out to have different quirks in different browsers.

Reply:
It's hard to find because nobody seems to care much about this, but my guess is Internet Explorer 7, Safari 5, etc.

Reply:
Ok, that's what I suspected. Sounds fine. Only other discrepancy to note for the future is that what we accept and display as a "login" might differ from asset/playlist permalinks.